iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/auth/entity/TablePrivilege.java

```diff
@@ -0,0 +1,166 @@
+    return this.grantOption;
+  }
+
+  public Set<Integer> getPrivilegesIntSet() {
+    Set<Integer> res = new HashSet<>();
+    for (PrivilegeType type : privileges) {
+      res.add(type.ordinal());
+    }
+    return res;
+  }
+
+  public Set<Integer> getGrantOptionIntSet() {
+    Set<Integer> res = new HashSet<>();
+    for (PrivilegeType type : grantOption) {
+      res.add(type.ordinal());
+    }
+    return res;
+  }
+
+  public void grantPrivilege(PrivilegeType priv) {
+    this.privileges.add(priv);
+  }
+
+  public void revokePrivilege(PrivilegeType priv) {
+    this.privileges.remove(priv);
+  }
+
+  public void grantOption(PrivilegeType priv) {
+    this.grantOption.add(priv);
+  }
+
+  public void revokeGrantOption(PrivilegeType priv) {
+    this.grantOption.remove(priv);
+  }
+
+  public void setPrivileges(int mask) {
+    final int PRI_SIZE = PrivilegeType.getPrivilegeCount(PrivilegeModelType.RELATIONAL);
+    for (int i = 0; i < PRI_SIZE; i++) {
+      if (((1 << i) & mask) != 0) {
+        this.privileges.add(AuthUtils.posToObjPri(i));
+        if (((1 << (i + 16)) & mask) != 0) {
+          this.grantOption.add(AuthUtils.posToObjPri(i));
+        }
+      }
+    }
+  }
+
+  public int getAllPrivileges() {
+    int privilege = 0;
+    for (PrivilegeType pri : privileges) {
+      privilege |= 1 << AuthUtils.objPriToPos(pri);
+    }
+    for (PrivilegeType pri : grantOption) {
+      privilege |= 1 << (AuthUtils.objPriToPos(pri) + 16);
+    }
+    return privilege;
+  }
+
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    TablePrivilege that = (TablePrivilege) o;
+    return tableName.equals(that.tableName)
+        && privileges.equals(that.privileges)
+        && grantOption.equals(that.grantOption);
+  }
+
+  public int hashCode() {
+    return Objects.hash(tableName, privileges, grantOption);
+  }
+
+  public String toString() {
+    StringBuilder builder = new StringBuilder();
+    builder.append(this.tableName).append("(");
+    List<PrivilegeType> privs = new ArrayList<>(this.privileges);
+    Collections.sort(privs);
+    for (PrivilegeType type : privs) {
+      builder.append(type);
+      if (grantOption.contains(type)) {
+        builder.append("_with_grant_option");
+      }
+      builder.append(",");
+    }
+    builder.append(")");
+    return builder.toString();
+  }
+
+  public void serialize(OutputStream outputStream) throws IOException {
+    ReadWriteIOUtils.write(this.tableName, outputStream);
+    SerializeUtils.serializePrivilegeTypeSet(this.privileges, (DataOutputStream) outputStream);
+    SerializeUtils.serializePrivilegeTypeSet(this.grantOption, (DataOutputStream) outputStream);
+  }
+
+  public void deserialize(ByteBuffer byteBuffer) {
+    this.privileges = new HashSet<>();
+    this.grantOption = new HashSet<>();
+    this.tableName = SerializeUtils.deserializeString(byteBuffer);
```

Comment:
Better use ReadWriteIOUtils.readString(Bytebuffer) here.. Though currently using this is also correct....